bindings/node/lib/index.js

```diff
@@ -0,0 +1,79 @@
+
+const native = require('../native');
+const stream = require('stream');
+const Promise = require('bluebird');
+
+class BufferStream extends stream.Duplex {
+    constructor(options) {
+        super(options);
+        this.inner = options.inner;
+    }
+
+    _read(size) {
+        let result = this.inner.read(size);
+        this.push(result);
+    }
+
+    _write(chunk, encoding, callback) {
+        this.inner.write(chunk, encoding);
+        callback(null);
+    }
+}
+
+function bufferRead(stream) {
+    return new Promise((resolve, reject) => {
+        let buffer = new BufferStream({ 
+            inner: new native.BufferStream() 
+        });
+        stream.on('error', reject);
+        stream.on('end', () => resolve(buffer.inner));
+
+        stream.pipe(buffer);
```

Comment:
@jhermsmeier This is probably a stupid question, but is / could this be a candidate for using Pipage? ;)